hedera-mirror-rest/transactions.js

```diff
@@ -320,6 +324,78 @@ const createTransferLists = (rows) => {
   };
 };
 
+/**
+ * Gets consensus_timestamps of the transactions that include a transfer from the staking reward account 800
+ * transfers are found, the staking_reward_transfers property will be added to the associated transaction object.
+ *
+ * @param transactions transactions list
+ * @return {[]|{number}[]}
+ */
+const getStakingRewardTimestamps = (transactions) => {
+  return transactions
+    .filter(
+      (transaction) =>
+        !_.isNil(transaction.crypto_transfer_list) &&
+        transaction.crypto_transfer_list.some(
+          (cryptoTransfer) => cryptoTransfer.entity_id === StakingRewardTransfer.STAKING_REWARD_ACCOUNT
+        )
+    )
+    .map((transaction) => transaction.consensus_timestamp);
+};
+
+/**
+ * Queries for the staking reward transfer list from the transfer list. If staking reward
+ * transfers are found, the staking_reward_transfers property will be added to the associated transaction object.
+ *
+ * @param transactions transactions list
+ * @return {Map<consensus_timestamp, staking_reward_transfer>} Map<ConsensusTimestamp, StakingRewardTransfer>
+ */
+const createStakingRewardTransferList = async (transactions) => {
+  const stakingRewardTimestamps = getStakingRewardTimestamps(transactions);
+  const rows = await getStakingRewardTransferList(stakingRewardTimestamps);
+  return convertStakingRewardTransfers(rows);
+};
+
+/**
+ * Queries for the staking reward transfer list
+ *
+ * @param {[]|{number}[]} stakingRewardTimestamps
+ * @return rows
+ */
+const getStakingRewardTransferList = async (stakingRewardTimestamps) => {
+  if (stakingRewardTimestamps.length === 0) {
+    return [];
+  }
+
+  const positions = _.range(1, stakingRewardTimestamps.length + 1).map((position) => `$${position}`);
+  const query = `
+    select ${StakingRewardTransfer.CONSENSUS_TIMESTAMP},
+           json_agg(json_build_object(
+             'account', ${StakingRewardTransfer.ACCOUNT_ID},
+             '${StakingRewardTransfer.AMOUNT}', ${StakingRewardTransfer.AMOUNT})) as staking_reward_transfers
+    from ${StakingRewardTransfer.tableName}
+    where ${StakingRewardTransfer.CONSENSUS_TIMESTAMP} in (${positions})
+    group by ${StakingRewardTransfer.CONSENSUS_TIMESTAMP}`;
+  const {rows} = await pool.queryQuietly(query, stakingRewardTimestamps);
+  return rows;
+};
+
+/**
+ * Convert db rows to staking_reward_transfer objects
+ * @param rows
+ * @returns {Map<any, any>}
+ */
+const convertStakingRewardTransfers = (rows) => {
+  const rewardsMap = new Map();
+  rows.map((t) => {
+    t.staking_reward_transfers.map((transfer) => {
+      transfer.account = EntityId.parse(transfer.account).toString();
+    });
+    rewardsMap.set(t.consensus_timestamp, t.staking_reward_transfers || []);
+  });
```

Comment:
```suggestion
  rows.forEach((t) => {
    t.staking_reward_transfers.forEach((transfer) => {
      transfer.account = EntityId.parse(transfer.account).toString();
    });
    rewardsMap.set(t.consensus_timestamp, t.staking_reward_transfers);
  });
```

Array.prototype.map() `creates a new array populated with the results of calling a provided function on every element in the calling array.`

Reply:
Thanks, this is now fixed.